src/chat.ts

```diff
@@ -35,6 +36,10 @@ export interface ChatOptions {
    * Defaults to 1024
    */
   maxOutputTokens?: number;
+
+  // Allows extra model specific parameters.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any;
```

Comment:
I'd rather we move this to a sub field

`extraParams: Record<string, unknown>`

Reply:
because otherwise it'll taint the whole interface

Reply:
These parameters are no different to the other ones, even at the API level. I don't think it makes sense to separate these out, but I'm also not sure what you mean by tainting the whole interface (not a TS expert). Can you elaborate?

Reply:
I mean something like this https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgPIAczAPYgM7IDeAUMspALbrRxgCuUEA-AFzIh0UBG0A3KcgDaAawgBPNnjBRQAcwC6bOCDH8AvsWIJcU5NnRsMWHcgC8RAZXRRaDCGwDkABgB0AVgcAaZAHofyCmA8PGoAG1CIABNvEGxyMWpkPDgYCDAxSwALIP1jEEjsCDxYyAAPILA2AEZvP3ZsMGRGOHCxZGU9TBwQdvzkAHcIB0Y9YQG4Nv7gMEzkaeINLR1sCJdQ7FkACn0ASiA

We just lose a lot of the type safety and ergonomics.

> Some models have extra parameters that we want users to access.

We can also type-safe by using a union of options that are model specific.

Reply:
We are ok with the lack of ability to catch typos. It'll throw an exception the moment they test it anyway. Lets go with the current approach.